Tools/Scripts/webkitpy/xcode/simulated_device.py

```diff
@@ -473,6 +483,13 @@ def initialize_devices(cls, requests, host=None, name_base='Managed', simulator_
         if len(requests):
             _log.debug(f'Running{"/specified" if udids else ""} simulators did not satisfy request. Finding matching non-booted ones, and/or creating new ones to satisfy the request.')
 
+            # New simulators will load all system frameworks into memory if the dyld shared cache doesn't exist.
+            # To avoid the immense strain this causes on the host's memory usage, we rebuild the cache here.
+            if len(SimulatedDeviceManager.INITIALIZED_DEVICES) == 0:
+                for runtime in SimulatedDeviceManager.AVAILABLE_RUNTIMES:
```

Comment:
We need to keep track of whether we've run this command. We don't want to run it twice in the same test run.

Also, is this a one-time-per-Xcode install thing? I'm actually a little surprised we need to do this every single run.

Reply:
> We need to keep track of whether we've run this command. We don't want to run it twice in the same test run.

Good point, I will add that.

> Also, is this a one-time-per-Xcode install thing? I'm actually a little surprised we need to do this every single run.

It's supposed to be. Anecdotal evidence indicates the shared cache can get into a bad state, which causes it not to be able to be used. Better to run it each time to be as safe as possible.